crates/artifact_content/src/file_system_state.rs

```diff
@@ -0,0 +1,71 @@
+use std::collections::{BTreeSet, HashMap};
+
+use crate::operation_text::hash;
+use isograph_config::PersistedDocumentsHashAlgorithm;
+
+type Filepath = String;
+type FileContent = String;
+type FileHash = String;
+
+#[derive(Debug, Clone)]
+pub struct FileSystemState {
+    state: HashMap<Filepath, (FileContent, FileHash)>,
+    pub sorted_keys: BTreeSet<String>,
+}
+
+impl FileSystemState {
+    pub fn new() -> Self {
+        Self {
+            state: HashMap::new(),
+            sorted_keys: BTreeSet::new(),
+        }
+    }
+
+    pub fn insert(&mut self, filename: Filepath, content: FileContent) {
+        let hashed_content = hash(&content, PersistedDocumentsHashAlgorithm::Sha256);
```

Comment:
https://stackoverflow.com/questions/2722943/is-calculating-an-md5-hash-less-cpu-intensive-than-sha-family-functions indicates that we should use md5 here

Reply:
In relay they use a user defined function that generate hashes, so one can use version control system to get file hash of existing file. I think is a good idea to have a enum here and associated hash algorithm so we could extend it later.

And I was thinking to use `blake3` for hashing files